### ISSUE TITLE
no-internal-modules doesn't work with aliases

### ISSUE THREAD

**Issue by jeron-diovis** [benmosher/eslint-plugin-import#1777]
With having following in eslint config:
```js
settings: {
  "import/resolver": {
    alias: {
      map: [
          [ "~", "./src" ],
          [ "components", "./src/components" ],
      ],
      extensions: [ ".js", ".jsx" ]
    }
  }
},

rules: {
  "import/no-internal-modules": [ "error", {
    "allow": [
      "components/*",
    ]
  }],
}
```
I have following results:
```sh
// denied
import Some from "components/Folder/Folder.js"

// allowed
import Some from "components/Folder"

// ! also allowed
import Some from "~/components/Folder/Folder"
```
Because path with "~" is recognized as unknown import type, and isn't validated at all. 
I expect this path to be resolved accordingly to configured aliases, and denied respectively. Or at least to be validated along with others, so I can explicitly add paths starting with `~` to `allow` array.

It is possible to implement?

**Comment by ljharb** [comment]
Do you perhaps need to set [this setting](https://github.com/benmosher/eslint-plugin-import#importinternal-regex)?

**Comment by jeron-diovis** [comment]
No, it doesn't help. Because, before this settings will be used, import must match `scopedRegExp` or `externalModuleRegExp`. Which aren't configurable.

https://github.com/benmosher/eslint-plugin-import/blob/3f46ccfe203fa5aad5767699df339a9b2f0294e3/src/core/importType.js#L65-L67

**Comment by ljharb** [comment]
Seems like a PR with test cases that adjusted line 67 there would do it :-)

**Comment by jeron-diovis** [comment]
Do you mean replacing `&&` with `||` after `matchesScopedOrExternalRegExp`?
As I don't know, is it intentional or kind of bug.

**Comment by ljharb** [comment]
I'm not quite sure either, to be honest! However, if it's intentional, tests will fail :-)

**Comment by dietergeerts** [comment]
```
// denied
import Some from "components/Folder/Folder.js"
```
Because your globs says it only allows for direct files:
```
  "components/*",
```
To allow direct files AND deep files, you need to do:
```
  "components/{*, **/*}",
```
I just encountered this myself.

**Comment by ljharb** [comment]
@jeron-diovis this seems answered; happy to reopen if not.

**Comment by jeron-diovis** [comment]
My question was about processing aliases. Not about how to allow deep files access.

**Comment by ljharb** [comment]
@jeron-diovis is there any chance you could provide a repo that reproduces this?

**Comment by jeron-diovis** [comment]
@ljharb https://github.com/jeron-diovis/eslint-internal-modules-demo

**Actual result:**
```sh
$ npm run lint

> eslint-import-aliases@1.0.0 lint
> eslint src/index.js

/Users/jeron_diovis/dev/tmp/eslint-import-aliases/src/index.js
  1:20  error  Reaching to "components/Folder/File.js" is not allowed  import/no-internal-modules

✖ 1 problem (1 error, 0 warnings)
```

**Expected result:**
Additional error at line 3, same as at line 1. Access to path `~/components/Folder/File.js` should be denied.
